src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkWildcardsCollector.kt

```diff
@@ -26,6 +26,14 @@ class SmkWildcardsCollector : SmkElementVisitor, PyRecursiveElementVisitor() {
     override val pyElementVisitor: PyElementVisitor
         get() = this
 
+    override fun visitSmkRule(rule: SmkRule) {
+        rule.getWildcardDefiningSection()?.let { visitSmkRuleOrCheckpointArgsSection(it) }
+    }
+
+    override fun visitSmkCheckPoint(checkPoint: SmkCheckPoint) {
+        checkPoint.getWildcardDefiningSection()?.let { visitSmkRuleOrCheckpointArgsSection(it) }
```

Comment:
Likely a wrong usage of visitor patter, I think should be `checkPoint.getWildcardDefiningSection()?.accept(this)` + refactored with visitSmkRule